docs/html/installing.rst

```diff
@@ -97,12 +97,7 @@ the `Python Packaging User Guide
 Upgrading pip
 =============
 
-On Linux or macOS::
-
- pip install -U pip
-
-
-On Windows [4]_::
+On Windows, Linux or macOS::
```

Comment:
Since this doesn't categorize anything (and implies that on other OSes the method would be different), I suggest replacing this by a filler, e.g. *in a shell* or something like that.

Reply:
I changed it to "In a shell". Thank you.

Reply:
Do we anywhere make the point that you don't always invoke Python using the `python` command? I feel as though we should clarify this *somewhere* if we're going to make `python -m pip` be the canonical way we say "run pip"...

(This is arguably off-topic for this PR, and honestly, arguably not even pip's problem to address, but I feel that without some such clarification, this change is attacking a symptom, rather than the cause, of the problem people have invoking pip).

Reply:
Adding some context on why we should use `python -m pip` should help. Should I create a new page or add it to an existing page? We can reference Brett Cannon's article the https://snarky.ca/why-you-should-use-python-m-pip/ there.

Reply:
Sorry, I wasn't clear. I'm fine with the `-m pip` bit. The problem is that on Windows, `python` probably isn't on `PATH`, you should use `py`. On Unix, you should often use `python3` rather than `python`. Unless you're in a virtual environment. Etc.

Using `-m pip` means that you don't need to think about how the `pip` command relates to whatever way you run Python. But it leaves you still needing to know how to invoke Python. We're using `python` as a shortcut for "whatever command you use to run Python" and I don't see anything explaining that (specifically "we don't necessarily mean you should type `python` literally")...

Reply:
I think there's also a need to mention that on Windows Python version is specified via `py -<major>.<minor>` instead of `python<major>.<minor>`.

Reply:
@pfmoore - Please let me know how to proceed? Does this change need further discussion before I proceed to making more changes?

Reply:
I'd like to know what other @pypa/pip-committers think.

Reply:
We can have a box on top of each page to tell the user they should use `pythonX.Y` or `py -X.Y` depending on the platform. But I don’t know how is best to annotate each code block to make it obvious to the user they should not copy-paste the command verbatim and must read the block at the top.